<launch>
    <arg name="xacro_path"   default="$(find ur5_allegro_moveit)/models/ur5_allegro.xacro" />
    <arg name="urdf_args"   default=""/>
    <arg name="load_moveit"   default="true"/>
    <arg name="sim"   default="true"/>
    <arg name="viz"   default="true"/>
    <arg name="start_leap"   default="true"/>
    <arg name="leap_topic"   default="/leap_motion/leap_filtered"/>
    <arg name="left_handed"   default="False"/>

    

    <include if="$(arg load_moveit)" file="$(find ur5_allegro_moveit)/launch/demo.launch">
        <arg name="rviz" value="false"/>
    </include>

    <!-- <group if="$(arg viz)"> -->
    <node name="teleop_rviz" pkg="rviz" type="rviz" if="$(arg viz)" args="-d $(find teleoperation_ur5_allegro_leap)/rviz/teleop_viz.rviz" />
    <!-- </group> -->
    
    <group if="$(arg start_leap)">
        <include file="$(find teleoperation_ur5_allegro_leap)/launch/leap_start.launch">
            <arg name="leap_camera_mode"   value="False"/>
            <arg name="leap_hand_root"   value="wrist"/>
            <arg name="leap_tf_hand_parent" value="leap_hands" />
            <arg name="leap_finger_parent" value="hand_root" />

            <arg name="leap_hand_tf" default="0 0 0 0.5000000532124816 0.5000000532124816 0.5 0.5000000532124816 world leap_hands"/>
            <arg name="leap_right_fingers_tf" value="0 0 0 0 0 0 1 hand_root right_leap_fingers"/>
            <arg name="leap_left_fingers_tf" value="0 0 0 0.5 0.5 0.5 0.5 world left_leap_fingers"/>
        </include>
    </group>

    <include file="$(find teleoperation_ur5_allegro_leap)/launch/ur5_relaxed_setup.launch">
            <arg name="teleop_mode_absolute" value="true"/>
            <arg name="sim" value="$(arg sim)"/>
    </include>
</launch>